### PR TITLE
Number overflow halting parser on crypto/file/contract update

### LINKED ISSUE #386: Long overflow when future expiration time
**Detailed Description**
If a transaction expiration time is set far enough in the future it can overflow the long we use to store nanosecond timestamps. This will then cause the record file to halt processing and the mirror node can never proceed. Currently, system accounts (<1000) have an expiration time of `31556889864403199` which is `12/31/1000000000 @ 11:59pm (UTC)`. Thus this scenario will occur if that field is ever updated in a CryptoUpdateAccount transaction, for example.

**Actual Behavior**
Long overflow and processing halts indefinitely

**Expected Behavior**
Handle the ArithmeticException and either put null or the max value that can fit without overflowing.

**Environment:**
- Version: v0.4.0-rc2

**Additional Context**
This impacts any transaction type that has a timestamp. Other timestamps like `consensusTimestamp` and `transactionValidStart` should not be impacted as they should always be in the past in order for the transaction to make it to the mirror node.

### PR DESCRIPTION
**Pull request by mike-burrage-hedera**
**Detailed description**:
- In Entities.create/update, handle overflows during conversion of (seconds,nanos) to a long nanos timestamp and set the database values accordingly.
- Utilities.convertToNanos changed due to its implementation adding an additional runtime exception possibility (DateTimeException), rather than just the expected ArithmeticException.
- Tested update via RecordFileLoggerCryptoTest and RecordFileLoggerFileTest

**Which issue(s) this PR fixes**:
Fixes #386 
Needs to be applied to `master` and `hcs`

**Special notes for your reviewer**:
Ran against the problematic testnet data:
```
2019-11-27 10:40:10,701 INFO  [scheduling-5] c.h.m.p.r.RecordFileParser Loading version 2 record file: 2019-11-16T21_50_31.218215Z.rcd
2019-11-27 10:40:10,702 ERROR [scheduling-5] c.h.m.u.Utility Long overflow when converting time to nanos timestamp : 17343941030s 648045000ns
2019-11-27 10:40:10,702 WARN  [scheduling-5] c.h.m.p.r.Entities Detected an 'update' transaction for entity 0.0.1769 with an invalid expiration date. 17343941030s 648045000ns
```

**Checklist**
- [ ] Documentation added
- [X] Tests updated

### CODE REVIEW

File: src/main/java/com/hedera/mirror/parser/record/Entities.java [hunk 1]
```diff
@@ -139,7 +139,21 @@ private long updateEntity(int fk_entity_type, long shard, long realm, long num,
             if ((exp_time_seconds != 0) || (exp_time_nanos != 0)) {
                 updateEntity.setLong(1, exp_time_seconds);
                 updateEntity.setLong(2, exp_time_nanos);
-                updateEntity.setLong(3, Utility.convertToNanos(exp_time_seconds, exp_time_nanos));
+                try {
```
[the remark this answers is not in the document]
Comment (steven-sheehy) [reply src/main/java/com/hedera/mirror/parser/record/Entities.java]: I agree. `Utility.convertToNanosWithFallback()` or something

File: src/test/java/com/hedera/mirror/util/UtilityTest.java [hunk 2]
```diff
@@ -232,6 +232,15 @@ void convertInstantToNanos(long seconds, int nanos) {
         );
     }
 
+    @ParameterizedTest(name = "with seconds {0} and nanos {1}")
+    @CsvSource({
+            "9223372036854775807, 0",
```
Comment (steven-sheehy) [inline src/test/java/com/hedera/mirror/util/UtilityTest.java]: Maybe add two test cases where it only overflows once you add the nanos

File: rest-api/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "rest-api",
-  "version": "0.4.0",
+  "version": "0.4.1-SNAPSHOT",
```
Comment (steven-sheehy) [inline rest-api/package.json]: Not sure why this shows as going from 0.4.0 to 0.4.1-SNAPSHOT. In `release/0.4` branch it is currently `0.4.1-rc`. Probably need to merge/rebase from that branch.

File: src/main/java/com/hedera/mirror/parser/record/Entities.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (steven-sheehy) [reply src/main/java/com/hedera/mirror/parser/record/Entities.java]: I added a `Utility.convertToNanosMax()`. Downside is we lose the log of which account caused the overflow, but we can always investigate that manually by looking for overflowed timestamps in the t_entities table.

File: rest-api/__tests__/integration.test.js [hunk 4]
```diff
@@ -391,6 +401,28 @@ test('DB integration test - transactions.reqToSql - Unknown transaction result a
   expect(res.rows[0].result).toEqual('UNKNOWN');
 });
 
+test('DB integration test - account min expiration time', async () => {
```
Comment (steven-sheehy) [inline rest-api/__tests__/integration.test.js]: We can remove these two test cases, right? The existing spec files show that the response is correct.

Comment (Nana-EC) [reply rest-api/__tests__/integration.test.js]: Removed

**Pull request merged**